tmail-backend/tmail-third-party/ai-bot/src/test/java/com/linagora/tmail/mailet/AiBotSuggestionMethodTest.java

```diff
@@ -0,0 +1,180 @@
+            .authenticate(BOB, PASSWORD)
+            .sendMessage(FakeMail.builder()
+                .name("name")
+                .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                    .setSender(BOB)
+                    .addToRecipient(ALICE)
+                    .setSubject("How can I cook an egg?")
+                    .setText("I do not know how to cook an egg. Please help me."))
+                .sender(BOB)
+                .recipient(ALICE));
+        imapClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(ALICE, PASSWORD)
+            .select(TestIMAPClient.INBOX)
+            .awaitMessageCount(Constants.awaitAtMostOneMinute, 1);
+
+        MimeMessage mimeMessage = MimeMessageUtil.mimeMessageFromString(imapClient.readFirstMessage());
+        System.out.println(Arrays.toString(mimeMessage.getHeader("Message-Id")));
+        assertThat(mimeMessage.getHeader("Sender")).containsOnly(BOB);
+        assertThat(mimeMessage.getHeader("Subject")[0]).isEqualTo("How can I cook an egg?");
+    }
+    @Test
+    public void shoudVerifyJmapMethod(GuiceJamesServer jamesServer) throws Exception {
+        jamesServer.getProbe(DataProbeImpl.class).fluent()
+            .addDomain(DEFAULT_DOMAIN)
+            .addUser(BOB, PASSWORD)
+            .addUser(ALICE, PASSWORD);
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(BOB, PASSWORD)
+            .sendMessage(FakeMail.builder()
+                .name("name")
+                .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                    .setSender(BOB)
+                    .addToRecipient(ALICE)
+                    .setSubject("How can I cook an egg?")
+                    .setText("I do not know how to cook an egg. Please help me."))
+                .sender(BOB)
+                .recipient(ALICE));
+        imapClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(ALICE, PASSWORD)
+            .select(TestIMAPClient.INBOX)
+            .awaitMessageCount(Constants.awaitAtMostOneMinute, 1);
+        MimeMessage mimeMessage = MimeMessageUtil.mimeMessageFromString(imapClient.readFirstMessage());
+
+        String request = "{\n" +
+            "  \"using\": [\"urn:ietf:params:jmap:core\", \"com:linagora:params:jmap:aibot\"],\n" +
+            "  \"methodCalls\": [\n" +
+            "    [\"AiBot/Suggest\", {\n" +
+            "      \"accountId\": \"ec77d4733cef70537b19b580db81fbbb6dc2a46a6d5662055cab568e75f71570\",\n" +
+            "      \"userInput\": \"explain to him how to cook an egg\",\n" +
+            "      \"emailId\": " + Arrays.toString(mimeMessage.getHeader("Message-Id")) + "\n" +
+            "    }, \"0\"]\n" +
+            "  ]\n" +
+            "}";
+        System.out.println("Generated Request: \n" + request);
```

Comment:
debug statement

Use .prettyPeek ou .log.all instead

(and do not commit it)